configure.ac

```diff
@@ -247,13 +247,9 @@ if test "$enable_avformat" = "yes"; then
     PKG_CHECK_MODULES(LIBAVFORMAT,  [libavformat libavcodec libavutil])
 fi
 
-#check openssl
-if test "$enable_md5" = "yes"; then
-    PKG_CHECK_MODULES([LIBBSD], [libbsd],
-        [AC_DEFINE([__ENABLE_MD5__], [1],
-            [Defined to 1 if MD5 API and --enable-md5[default] are enabled])],
-        [])
-fi
+
+PKG_CHECK_MODULES([BSD], [libbsd], [have_bsd="yes"], [have_bsd="no"])
+AM_CONDITIONAL(HAVE_LIBBSD, test x"$have_bsd" = "xyes")
```

Comment:
AM_CONDITIONAL only defines it for makefiles to use.  AC_DEFINE defines it in config.h for sourcecode to use.

It seems we need two defines for sourcecode... one if --enable-md5 is present and one if libbsd is present.

We already have ENABLE_MD5 for makefiles

Reply:
I want to change the logical here.
We can autodetect libbsd, if we do not have libbsd we can use built-in md5 sum. 
this logical should work fine in env with or without libbsd.